### ISSUE TITLE
Markers add/set should accept range

### ISSUE THREAD

**Issue by pjasiun** [ckeditor/ckeditor5-engine#761]
I should not be forced to transform range to the live range whenever I want to create a marker.
I should not need to care about destroying live range when I'm using markers.
I should be able to create a marker based on the selection, these APIs should be compatible: `doc.markers.add( name, doc.selection.getFirstRange() );`

**Comment by pjasiun** [comment]
I fact, I think that people using markers may have no idea what `LiveRange` is.

**Comment by pjasiun** [comment]
And even if the marker is created based on the `LiveRange` it should clone it and handle internally. Otherwise, there might be problems with destroying.

**Comment by scofalik** [comment]
**tl;dr:** I agree that `Range` should be transformed to `LiveRange`. I don't agree about cloning `LiveRange`s though. `LiveRange`s that are created by `MarkerCollection` (upon adding them to it) should be then destroyed by `MarkerCollection` when the collection is destroyed. Explanation below.

-----

The idea to force `LiveRange` seemed good at first, for two reasons:

1. If all ranges passed to `MarkerCollection` are `LiveRange`s, `MarkerCollection` does not need to care about destroying ranges -- code that created the `LiveRange` should `detach` it.
2. It raises awareness. `LiveRange` is a costly resource. `LiveRange` creates event listener that is fired every time there is an action in the editor: selection gets moved, letter is typed, etc. So in this case, I think it's better that users exactly know what happens.

However, when I was using `MarkersCollection`, I also came to conclusion that it is cumbersome to import `LiveRange` every time and use `LiveRange.createFromRange( myRange )`. So we need some kind of solution.

1. I am fine with automatic `Range` to `LiveRange` conversion.
2. I am not fine with cloning `LiveRange` when it's passed `#add` method. This is for efficiency reasons. It creates twice as many `LiveRange`s. It is also annoying when you debug stuff in `LiveRange` transformation, because when you set a breakpoint inside transformation function you get there multiple times. Currently, when I debugged some stuff there I already got annoyed because `LiveRange`s are also in `LiveSelection`. It's hard to say "which" `LiveRange` you are transforming at the moment, etc. Long story short: it's better to have minimal amount of `LiveRange`s.

**Comment by pjasiun** [comment]
The problem is destroying these live ranges. If `MarkerCollection` reuses the live range, you can not remove it as soon as you do not need it, because `MarkerCollection` may still need it. It means that it's hard to guess when you can destroy it and you won't destroy it, just in case, to avoid bugs.

At the same time, I think that live ranges should not be created to be passed, they should be created for "special tasks", to track some changes, as a tool to do some a job, not as something you pass to methods, as ranges. This is why I think that markers should not accept live ranges as a parameter.

Also, I still think that a user who uses markers should not need to know what live range is. He creates a marker to mark something and remove it when it is not needed anymore. We may aware him of the fact that creating too many markers might be bad for the performance because they are tracking changes. We can also mention in docs that we have also `LiveRanges`, they are very similar, and `LiveRanges` might be what he really need instead of markers. But that's all he needs to know. He does not need to know that markers are based on live ranges (that's ok to inform him about it, but not to expect that he will understand consequences).  For sure he should not need to care about detaching live range when he delete marker.
